numpy/lib/_function_base_impl.py

```diff
@@ -5164,22 +5156,21 @@ def trapz(y, x=None, dx=1.0, axis=-1):
     return ret
 
 
-# __array_function__ has no __code__ or other attributes normal Python funcs we
-# wrap everything into a C callable. SciPy however, tries to "clone" `trapz`
-# into a new Python function which requires `__code__` and a few other
-# attributes. So we create a dummy clone and copy over its attributes allowing
-# SciPy <= 1.10 to work: https://github.com/scipy/scipy/issues/17811
-assert not hasattr(trapz, "__code__")
```

Comment:
This is a regular cleanup, not directly related to `trapz`/`trapezoid`. It's a workaround that is simply no longer needed; it's been fixed in SciPy for a while, and older releases don't support numpy 2.0